fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -168,18 +204,38 @@ public static CompoundTag toTag(boolean isClientSync) {
 					registryTag.putInt(id.toString(), rawId);
 				}
 
+				// Look for existing registry key/values that are not in the current registries
+				if (!isClientSync && existingRegistryData != null) {
+					for (String key : existingRegistryData.getKeys()) {
+						if (!registryTag.contains(key)) {
+							LOGGER.info("Saving orphaned registry entry: " + key);
+							registryTag.putInt(key, registryTag.getInt(key));
+						}
+					}
+				}
+
 				mainTag.put(registryId.toString(), registryTag);
 			}
 		}
 
+		// Ensure any orphaned registry's are kept on disk
+		if (!isClientSync && activeIdMap != null) {
+			for (String registryKey : activeIdMap.getKeys()) {
+				if (!mainTag.contains(registryKey)) {
+					LOGGER.info("Saving orphaned registry: " + registryKey);
+					mainTag.put(registryKey, activeIdMap.getCompound(registryKey));
+				}
+			}
+		}
+
 		CompoundTag tag = new CompoundTag();
 		tag.putInt("version", 1);
```

Comment:
Should probably bump version here

Reply:
No, it should be compatible with previous fabric api versions. I need to test this.